spring-boot-samples/spring-boot-sample-kafka/src/test/java/sample/kafka/SampleKafkaApplicationTests.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sample.kafka;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.rule.OutputCapture;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Integration tests for demo application.
+ *
+ * @author hcxin
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class SampleKafkaApplicationTests {
```

Comment:
This test does not pass if Kafka is not running (which is something that should be expected). Can you review the documentation as it explains how you can embed the broker in the test.

Reply:
Hi, i'm sorry for replying you late, i have updated the pr.
1) Add KafkaEmbedded to embed the broker in the test.
2) Using JSON to serialize the POJO's content (JsonSerializer and JsonDeserializer).

I have tested on mac and windows platforms, it works well, thanks.